pkg/apis/build/v1beta1/build_conversion.go

```diff
@@ -48,6 +48,14 @@ func (src *Build) ConvertTo(ctx context.Context, obj *unstructured.Unstructured)
 		alphaBuild.ObjectMeta.Annotations[v1alpha1.AnnotationBuildRunDeletion] = strconv.FormatBool(*src.Spec.Retention.AtBuildDeletion)
 	}
 
+	// convert OCIArtifact to Bundle
+	if src.Spec.Source.OCIArtifact != nil {
+		alphaBuild.Spec.Source.BundleContainer = &v1alpha1.BundleContainer{
+			Image: src.Spec.Source.OCIArtifact.Image,
+			Prune: (*v1alpha1.PruneOption)(src.Spec.Source.OCIArtifact.Prune),
+		}
+	}
```

Comment:
You need to convert `src.Spec.Source.OCIArtifact.PullSecret` to `alphaBuild.Spec.Source.Credentials.Name`.

Reply:
I believe that's taken care of at [L391](https://github.com/shipwright-io/build/blob/main/pkg/apis/build/v1beta1/build_conversion.go#L391-L393). Let me know in case I missed anything.
The e2e tests are no longer failing. I missed out on specifying the type as ociArtifact for source in the `SourceBundle()` due to which default case was being picked up in the switch case.